yadll/tensor/optimizers/optimizer.py

```diff
@@ -0,0 +1,26 @@
+from abc import ABC, abstractmethod
+import numpy as np
+
+
+class Optimizer(ABC):
+    def __init__(self, params):
+        self.params = params
+
+    @abstractmethod
+    def step(self):
+        pass
+
+    def zero_grad(self):
+        for p in self.params:
+            p.grad = None
+
+
+class SGD(Optimizer):
+    def __init__(self, params, lr: float, momentum: float):
```

Comment:
I added momentum here so that the args count is the same, but it isn't actually doing anything.
We need to figure out why the ``test_sgd_momentum`` unit-test is passing